src/wazuh_modules/vulnerability_detector/wm_vuln_detector.c

```diff
@@ -2012,6 +2019,52 @@ int wm_vuldet_oval_discard_mismatching_cve_entries(sqlite3 *db, agent_software *
     return ret;
 }
 
+int wm_vuldet_oval_prescan_SUSE_dependencies(sqlite3 *db, agent_software *agents_it) {
+    sqlite3_stmt    *stmt = NULL;
+    sqlite3_stmt    *stmt_update = NULL;
+    char            *operation;
+    char            *operation_value;
+    char            *version;
+    char            *pkg_name;
+    int             result;
+    int             ret = 0;
+
+    if (wm_vuldet_prepare(db, vu_queries[VU_GET_SUSE_DEPS], -1, &stmt, NULL) != SQLITE_OK) {
+        return wm_vuldet_sql_error(db, stmt);
+    }
+    sqlite3_bind_text(stmt, 1, agents_it->agent_id, -1, NULL);
+
+    while (result = wm_vuldet_step(stmt), result == SQLITE_ROW) {
+        version = (char *)sqlite3_column_text(stmt, 0);
+        operation_value = (char *)sqlite3_column_text(stmt, 1);
+        pkg_name = (char *)sqlite3_column_text(stmt, 2);
+        operation = (char *)sqlite3_column_text(stmt, 3);
+
+        // Compare the dependency's version against the one in the feed to set the flag if applicable
+        ret = wm_checks_package_vulnerability(version, operation, operation_value, VER_TYPE_RPM);
+        if (ret != VU_NOT_VULNERABLE && ret != OS_INVALID) {
```

Comment:
To avoid showing logs when you try to update the installed flag with versions that have different targets, it is necessary to compare that the _agent's target_ is `equal` to the _dependency target_.

To do this, for example you can add the target field to the `VU_GET_SUSE_DEPS` query, and compare that result with the variable `vu_feed_tag ​[agents_it-> dist_ver]`